src/main.rs

```diff
@@ -268,3 +275,324 @@ async fn main() {
     axum::serve(listener, start_seed_registry()).await.unwrap();
     println!("Usage: {} <sha256sum>", args[1]);
 }
+
+#[cfg(test)]
+mod test {
+    use anyhow::{Context as _, Ok, Result, bail};
+    use bollard::{Docker, image::CreateImageOptions};
+    use flate2::{Compression, write::GzEncoder};
+    use futures_util::{TryStreamExt, future::ready};
+    use regex_lite::Regex;
+    use sha2::{Digest, Sha256};
+    use std::{
+        fs::File,
+        io::{BufRead, BufReader, Cursor, Seek, Write},
+        path::{Path, PathBuf},
+    };
+
+    use crate::{start_seed_registry, unpack};
+
+    struct TempDir {
+        path: PathBuf,
+    }
+
+    impl TempDir {
+        fn new() -> Result<Self> {
+            let path = std::env::temp_dir().join(format!("zarf-test-{}", std::process::id()));
+            std::fs::create_dir_all(&path).context("should have created temporary directory")?;
+            Ok(Self { path })
+        }
+
+        fn path(&self) -> &PathBuf {
```

Comment:
nit - this is returning a reference to the concrete type - want to check if this is intentional as it _may_ be more idiomatic to return a borrowed type

```suggestion
        fn path(&self) -> &Path {
```